cxx/gendb.hh

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+#include <map>
+#include <random>
+#include <string>
+
+#include "distributions/crp.hh"
+#include "hirm.hh"
+#include "observations.hh"
+#include "pclean/schema.hh"
+#include "pclean/schema_helper.hh"
+
+class GenDB {
+ public:
+  const PCleanSchema& schema;
+
+  // TODO(emilyaf): Merge PCleanSchemaHelper and GenDB.
+  PCleanSchemaHelper schema_helper;
+
+  // This data structure contains entity sets and linkages. Semantics are
+  // map<class_name, map<primary_key, map<reference_field_name, ref_val>>>,
+  // where primary_key and ref_val are (integer) entity IDs.
+  std::map<std::string, std::map<int, std::map<std::string, int>>>
+      reference_values;
+
+  HIRM* hirm;
+
+  // CRPs for latent entities, where the "tables" are entity IDs and the
```

Comment:
I would reverse the order of these two sentences when explaining domain_crps:  "domain_crps[class_name] gives the CRP used to generate entity ID's for that class.  Inside that CRP, the `tables` are entity IDs and the `customers` are unique identifiers of observations of that class."

(Basically, explain the most important thing first and then explain the confusing details!)

Reply:
Agreed, done.